### PR TITLE
Replace Request with Axios

### PR DESCRIPTION
**Pull request by cbardini**

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Version History
 
+## 1.0.19 / 2021-05-04
+
+- Replace depricated package Request with Axios
```
Comment (dcbroad3) [inline HISTORY.md:5]: typo, `deprecated`

File: SdsClient.js [hunk 2]
```diff
@@ -64,31 +66,30 @@ module.exports = {
...
-          return restCall({
-            url: authority,
-            method: 'POST',
+          var body = new URLSearchParams({
+            grant_type: 'client_credentials',
+            client_id: clientId,
+            client_secret: clientSecret,
+          });
+
+          console.log(body.toString());
```
Comment (dcbroad3) [inline SdsClient.js:87]: This looks like it should've only been there for debugging. Is this intended to be part of the sample?

File: SdsClient.js [hunk 3]
```diff
@@ -547,61 +533,68 @@ module.exports = {
...
         method: 'DELETE',
         headers: this.getHeaders(),
-        gzip: true,
       });
     };
 
+    this.gzipCompress = function (data, headers) {
+      if (
+        'Content-Encoding' in headers &&
+        headers['Content-Encoding'] == 'gzip'
```
Comment (dcbroad3) [inline SdsClient.js:588]: It is generally preferred to use `===` in JavaScript for equality checks. Might also want to consider using lowercase on the header, e.g.:
`headers['Content-Encoding'].toLowerCase() === 'gzip'`

File: package.json [hunk 4]
```diff
@@ -17,11 +17,10 @@
   },
   "requires": true,
   "dependencies": {
-    "request-promise": "^4.2.6"
+    "axios": "^0.21.1"
```
Comment (dcbroad3) [inline package.json:20]: Should also increment the sample version inside this file as well as `package-lock.json`

File: SdsClient.js (hunk 2 above)
Comment (cbardini) [reply SdsClient.js:87]: fixed typo

File: SdsClient.js (hunk 3 above)
Comment (cbardini) [reply SdsClient.js]: That's some interesting syntax. I have updated the line to your recommendation.

File: package.json (hunk 4 above)
Comment (cbardini) [reply package.json:20]: versions update in the package.json and package-lock.json files

**Review by dcbroad3 — APPROVED** [review]

**Review by kelseyduffy — APPROVED** [review]

**Pull request merged**